consensus/core/src/transaction_certifier.rs

```diff
@@ -148,10 +148,9 @@ impl TransactionCertifier {
                     } else {
                         // Own votes are needed for blocks not yet included in own blocks. They will be
                         // added to proposed blocks together.
-                        let reject_transaction_votes = block_verifier
-                            .verify_and_vote(b.signed_block())
-                            .unwrap_or_else(|e| {
-                                panic!("Failed to verify block during recovery: {}", e)
+                        let reject_transaction_votes =
+                            block_verifier.vote(&b).unwrap_or_else(|e| {
```

Comment:
@mwtian let me know if we had _any_ reason to re-verify the block it self (I don't think so)

Reply:
Since this is the recovery path, the original thinking is that it is ok to re-run verification at least as a sanity check, and not worth it to add a new API for this case. I plan to refactor this logic to minimize the number of blocks loaded to the last few commits. Either way is fine.